app/services/resource_service.rb

```diff
@@ -108,4 +156,10 @@ def set_default_categories
       @resource.categories << category unless @resource.category_ids.include?(cat_id.to_i)
     end
   end
+
+  def is_singular?(str)
+    str = str.to_s
```

Comment:
`str` heißt string? und dann passiert string to string? bisschen eigenartig. wenn es nicht sicher ein string ist, was hier reinkommt, dann sollte es nicht `str` heißen, oder?

Reply:
Variablennamen angepasst